src/main.cpp

```diff
@@ -0,0 +1,102 @@
+#include <string>
+#include <iostream>
+#include <map>
+#include <fstream>
+#include <sstream>
+#include "CityExplorer.h"
+
+const std::string DEFAULT_DATA_PATH = "data.txt";
+
+void initExplorer(CityExplorer& ce);
+void printResults(const std::multimap<Coordinate, std::string>& cities);
+
+int main() {
+
+	try
+	{
+		std::string command = "";		
+		float radius = 0;
+		int norm = 0;	
+
+		CityExplorer explorer;
+		initExplorer(explorer);
+		std::multimap<Coordinate, std::string> closeCities = {};
+		
+		std::cout << "Please enter selected city name (with line break after it):\n";
+		std::getline(std::cin, command);
+
+		while (command != "0") {
+			while (!explorer.cityExists(command)) {
+				std::cout << "City doesn't exist.\n";
+				std::cout << "Please enter selected city name (with line break after it):\n";
+				std::getline(std::cin, command);
+			}
+			std::cout << "Please enter the wanted radius:\n";
+			std::cin >> radius;
+			while (radius < 0) {
+				std::cout << "Invalid radius.\n";
+				std::cout << "Please enter the wanted radius:\n";
+				std::cin >> radius;
+			}
+			std::cout << "Please enter the wanted norm (0 - L2, Euclidean distance, 1 - Linf, Chebyshev distance, 2 - L1, Manhattan distance):\n";
+			std::cin >> norm;
+			while (norm < 0 || norm > 2) {
+				std::cout << "Invalid norm number.\n";
+				std::cout << "Please enter the wanted norm (0 - L2, Euclidean distance, 1 - Linf, Chebyshev distance, 2 - L1, Manhattan distance):\n";
+				std::cin >> norm;
+			}
+			closeCities = explorer.explore(command, radius, norm);
+			std::cout << "Number of northern cities: " << explorer.northernCities(command) << std::endl;
+			std::cout << "Total number of cities: " << explorer.getCityCount() << std::endl;
```

Comment:
you should return the count on the cities in the radius, not the cities in the file